### PR TITLE
refactor: split flightsql crate into smaller modules

### PR DESCRIPTION
**Pull request by alamb**
I am following up with @tustvold's suggestion in https://github.com/influxdata/influxdb_iox/pull/6667#discussion_r1086468817 but I am splitting the code into smaller modules for review.

# Changes:
1. Break up flightsql into smaller modules. No behavior change is intended

### CODE REVIEW

File: flightsql/src/cmd.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+/// Encode a PreparedStatementHandle as Bytes
+impl From<PreparedStatementHandle> for Bytes {
+    fn from(value: PreparedStatementHandle) -> Self {
+        Bytes::from(value.query.into_bytes())
+    }
+}
+
+/// Decoded /  validated FlightSQL command messages
+#[derive(Debug, Clone)]
+pub enum FlightSQLCommand {
```
Comment (tustvold) [inline flightsql/src/cmd.rs:52]: I think this should be split into `FlightSQLCommand` and `FlightSQLAction` but don't feel strongly

I also wonder if the `FlightSQL` prefix is redundant, given this is in a module called flightsql, but again don't feel strongly

File: flightsql/src/error.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    #[snafu(display("Unsupported FlightSQL message type: {}", description))]
+    UnsupportedMessageType { description: String },
+
+    #[snafu(display("Protocol error. Method {} does not expect '{:?}'", method, cmd))]
+    Protocol { cmd: String, method: &'static str },
+}
+
+pub type Result<T, E = Error> = std::result::Result<T, E>;
+
+impl From<FlightError> for Error {
```
Comment (tustvold) [inline flightsql/src/error.rs:41]: FWIW I think snafu can auto-derive these if you do something like `#[snafu(display("{}", source), context(false))]`

**Review by tustvold — APPROVED** [review]

File: flightsql/src/cmd.rs (hunk 1 above)
Comment (alamb) [reply flightsql/src/cmd.rs:52]: > I think this should be split into FlightSQLCommand and FlightSQLAction but don't feel strongly

This makes the implict (and possibly correct) assumption that `Action..` is only passed to do_action and `Command...` is only passed to `do_get`. 

I would like to postpone this until after I have implemented parameters as I am not really sure

File: flightsql/src/error.rs (hunk 2 above)
Comment (alamb) [reply flightsql/src/error.rs:41]: Will look into it

Comment (alamb) [reply flightsql/src/error.rs]: 5bee65783 👍

**Pull request merged**